framework/include/restart/DataIO.h

```diff
@@ -123,6 +126,13 @@ template<typename T>
 inline void
 dataStore(std::ostream & stream, T & v, void * /*context*/)
 {
+  #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
+    static_assert( std::is_polymorphic<T>::value == false,
+      "Cannot serialize a class that has virtual members!\nWrite a custom dataStore() template specialization!\n\n");
+    static_assert( std::is_trivially_copyable<T>::value || std::is_same<T, Point>::value,
```

Comment:
It's useful, so you could use it for discovering issues of this type, but I don't think we can leave it on all the time, as it will prevent legit apps from compiling, right?  Do we have any structs-containing-Points currently in the apps?  How did you run across this issue in the first place?